files/buildspec_tf_plan.yml.tftpl

```diff
@@ -29,5 +31,11 @@ phases:
 
 artifacts:
   files:
-    - thePlan.tfp
+    %{ if EXTRA_FILE != "" }
+      - ${EXTRA_FILE}
+      - thePlan.tfp
+      %{ else }
+      - thePlan.tfp
+      %{ endif }
```

Comment:
Is there need for the `%{ else }` branch?
```
files:
    - thePlan.tfp
    %{ if EXTRA_FILE != "" }
    - ${EXTRA_FILE}
    %{ endif }
```

Reply:
Before switching to `set(string)` we need to make sure that this templating scheme supports foreach-type of loops

Reply:
> Is there need for the `%{ else }` branch?
> 
> ```
> files:
>     - thePlan.tfp
>     %{ if EXTRA_FILE != "" }
>     - ${EXTRA_FILE}
>     %{ endif }
> ```

No, I bet this will work as well.